sys/include/riot_hdr.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_riot_hdr RIOT header helpers and tools
+ * @ingroup     sys
+ * @{
+ *
+ * # RIOT image header for riotboot bootloader
```

Comment:
This should not mention `riotboot` or explain how to compile it as the dependency is in the other order.
It would be really visible by having commits in the right order.

Reply:
I think this needs to be documented somewhere and riot_hdr doesn't make sense without riotboot using it here. True, riot_hdr can be used by other bootloaders but for now the usage is here.

I agree the commits need reorder as I said before, however I think this is a separated issue, as long as the usage of riot_hdr is showcased somewhere.